components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -413,7 +414,8 @@ private void validateCallbackURI(OAuthConsumerAppDTO application) throws Identit
         boolean isCallbackUriRequired = application.getGrantTypes().contains(AUTHORIZATION_CODE) ||
                 application.getGrantTypes().contains(IMPLICIT);
 
-        if (isCallbackUriRequired && StringUtils.isEmpty(application.getCallbackUrl())) {
+        if (isCallbackUriRequired && (StringUtils.isEmpty(application.getCallbackUrl())
+                && application.getCallBackUrls().isEmpty())) {
```

Comment:
```suggestion
        if (isCallbackUriRequired && (StringUtils.isEmpty(application.getCallbackUrl())
                || application.getCallBackUrls().isEmpty())) {
```
This should be OR right?